src/vcpkg/configure-environment.cpp

```diff
@@ -10,57 +10,107 @@
+            }
+        }
+
+        if (needs_provisioning)
+        {
+            msg::println(msgDownloadingVcpkgCeBundle, msg::version = VCPKG_BASE_VERSION_AS_STRING);
             const auto ce_uri =
                 "https://github.com/microsoft/vcpkg-tool/releases/download/" VCPKG_BASE_VERSION_AS_STRING
                 "/vcpkg-ce.tgz";
             const auto ce_tarball = paths.downloads / "vcpkg-ce-" VCPKG_BASE_VERSION_AS_STRING ".tgz";
-            download_manager.download_file(fs, ce_uri, ce_tarball, std::string(MACRO_TO_STRING(VCPKG_CE_SHA)));
-#else  // ^^^ VCPKG_CE_BUNDLE_SHA / !VCPKG_CE_BUNDLE_SHA vvv
-            print2(Color::warning, "Downloading latest vcpkg-ce bundle\n");
-            const auto ce_uri = "https://github.com/microsoft/vcpkg-tool/releases/latest/download/vcpkg-ce.tgz";
-            const auto ce_tarball = paths.downloads / "vcpkg-ce-latest.tgz";
-            download_manager.download_file(fs, ce_uri, ce_tarball, nullopt);
-#endif // ^^^ !VCPKG_CE_BUNDLE_SHA
-            auto npm_path = node_root / "node_modules" / "npm" / "bin" / "npm-cli.js";
-            if (!fs.exists(npm_path, VCPKG_LINE_INFO))
-            {
-                npm_path = Path(node_root.parent_path()) / "lib" / "node_modules" / "npm" / "bin" / "npm-cli.js";
-            }
-
-            Command cmd_provision(node_path);
-            cmd_provision.string_arg(npm_path);
-            cmd_provision.string_arg("--force");
-            cmd_provision.string_arg("install");
-            cmd_provision.string_arg("--no-save");
-            cmd_provision.string_arg("--no-lockfile");
-            cmd_provision.string_arg("--scripts-prepend-node-path=true");
-            cmd_provision.string_arg("--silent");
-            cmd_provision.string_arg(ce_tarball);
-            const auto provision_status = cmd_execute(cmd_provision, WorkingDirectory{paths.root}, env);
-            fs.remove(ce_tarball, VCPKG_LINE_INFO);
-            if (provision_status != 0)
-            {
-                fs.remove_all(node_modules, VCPKG_LINE_INFO);
-                Checks::exit_with_message(VCPKG_LINE_INFO, "Failed to provision vcpkg-ce.");
-            }
+            download_manager.download_file(fs, ce_uri, ce_tarball, VCPKG_CE_SHA_AS_STRING);
+            extract_ce_tarball(paths, ce_tarball, node_path, node_modules);
+            fs.write_contents(ce_sha_path, VCPKG_CE_SHA_AS_STRING, VCPKG_LINE_INFO);
         }
+#else  // ^^^ VCPKG_CE_SHA / !VCPKG_CE_SHA vvv
+        fs.remove(ce_sha_path, VCPKG_LINE_INFO);
+        fs.remove_all(ce_path, VCPKG_LINE_INFO);
+        msg::println(Color::warning, msgDownloadingVcpkgCeBundleLatest);
+        const auto ce_uri = "https://github.com/microsoft/vcpkg-tool/releases/latest/download/vcpkg-ce.tgz";
+        const auto ce_tarball = paths.downloads / "vcpkg-ce-latest.tgz";
+        download_manager.download_file(fs, ce_uri, ce_tarball, nullopt);
+        extract_ce_tarball(paths, ce_tarball, node_path, node_modules);
```

Comment:
Doesn't this make it so that when building from source you have to download ce each time?

Reply:
> Doesn't this make it so that when building from source you have to download ce each time?

That's correct.

Reply:
Ok, if this is intended then. Isn't it going to affect users in "exotic platforms"?

Reply:
> Ok, if this is intended then. Isn't it going to affect users in "exotic platforms"?

Yes, but given that we have no means of determining out-of-date-ness I don't have a better option. The alternative would be to never update things and hope the user nukes node_modules when they update which seems worse.

Of course, once https://github.com/microsoft/vcpkg-tool/pull/428 happens the exotics are going to have to build the ce parts themselves so...

Also note that we adopted this Wednesday that we effectively "don't care" about the exotics on the basis that "if it isn't tested it doesn't exist".